### ISSUE TITLE
Modify constraints API to support creating foreign key constraints

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#942]
## Current behavior

- I can't create FK constraints.
- I can create a _unique_ constraint with a POST to `/tables/{table_id}/constraints/` and `type: "unique"`.

## Desired behavior

- Creating FK constraints should function similarly to creating unique constraints.

## Notes
- The API for creating unique constraints is currently in flux due to #898. We should be sure to follow the same pattern for FK constraints, which will likely mean requiring the API request to specify all tables and columns by id instead of name.
- We don't plan to ever support _updating_ FK constraints. If the user wants to change the constraint, they'll need to drop it and add a new one.
- If it is fairly quick to implement, the API should support as many of the following features as it can, even though there's no UI for them yet. If you do not support these in your PR, please design the API in a way that can be easily extended to support them in the future without breaking changes. It would also be helpful to create separate placeholder issues to track any issues not handled by the PR.
    - Creating deferred FK constraints
    - Setting the `ON DELETE` attribute to any of the five possible actions.
    - Setting the `ON UPDATE` attribute to any of the five possible actions.
    - Creating multi-column FKs, including the ability to pass `MATCH FULL`.

## Additional context
- Blocked by API definition in #943
- [Foreign Key Spec](https://wiki.mathesar.org/en/design/specs/foreign-key-constraints) -- Scenario 1
- [Postgres docs for Constraints](https://www.postgresql.org/docs/13/ddl-constraints.html)
- [Postgres docs for `SET CONSTRAINTS`](https://www.postgresql.org/docs/13/sql-set-constraints.html)
- The UI only supports single-column FKs, but we might want to add support for creating multi-column FKs in the future ([discussion](https://matrix.to/#/!xInuTkBwjXZXYatIlm:matrix.mathesar.org/$ZMwTJrIU7sWuqtrSndMHaRZ3hR2rAW9A8JEemqQOW6w?via=matrix.mathesar.org&via=matrix.org)).

**Comment by seancolsen** [comment]
@kgodey I'm assigning this to you to help answer the above questions and loop others in as needed.

**Comment by kgodey** [comment]
@seancolsen I've updated the ticket description to resolve questions. I'm going to mark the ticket as blocked until #898 is resolved and the API for constraints stabilizes.
